oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/db/OSHDBIgnite.java

```diff
@@ -40,7 +39,6 @@ public OSHDBIgnite() {
 
   public OSHDBIgnite(Ignite ignite) {
     this.ignite = ignite;
-    this.ignite.cluster().state(ClusterState.ACTIVE_READ_ONLY);
```

Comment:
We should actually check the current cluster().state() and throw an exception right here if it is not in active or active_read_only state.
Because in a case when the cluster is actually not active, there will be probably an exception on the first cache operation somewhere which would then be unexpected. 
I think it's better to notify as soon as possible.

Reply:
can you please suggest an implementation for this?

Reply:
PS: what if the cluster goes active/inactive between the time where the connection gets established and when the actual OSHDB query is performed?

Reply:
> can you please suggest an implementation for this?

sure here are 2 suggestions:

```java
  // explicit check for good states
  if (!EnumSet.of(ACTIVE, ACTIVE_READ_ONLY).contains(cluster.state())) {
    // throw exception about cluster is not ready (yet)! 
  }
  // or check for the fail state
  if (cluster.state() == INACTIVE) {
    // throw exception about cluster is not ready (yet)! 
  }
```
I think the first suggestion is more robust in case there will be in the future more and other ClusterStates

Reply:
> PS: what if the cluster goes active/inactive between the time where the connection gets established and when the actual OSHDB query is performed?

When it goes inactive while a OSHDB query is performed, probably on the next cache operation there could be an exception similar to 
> class org.apache.ignite.internal.processors.cache.distributed.dht.IgniteClusterReadOnlyException: Failed to finish operation. Cluster in read-only mode!

We can check what specific exception will be thrown if any at all. I will try it in the test cluster. And report it back here

Reply:
For reference: will be implemented in #337